src/MetaDataConverterMain.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright 2018, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Johannes Kalmbach (johannes.kalmbach@gmail.com)
+//
+#include "./index/MetaDataConverter.h"
+#include <iostream>
+#include "./global/Constants.h"
+
+// _________________________________________________________
+int main (int argc, char** argv) {
+  if (argc != 2) {
+    std::cerr << "Usage: ./MetaDataConverterMain <originalFilename>\n";
+    exit(1);
+  }
+  std::string in = argv[1];
+  std::string out;
+  std::string mmap;
+  if (in.size() < 4) {
+    std::cerr << "error, infile must be called xxx.ops or xxx.osp\n";
+    exit(1);
+  } else {
+    std::string suffix = in.substr(in.size() - 4);
+    if ((suffix != ".ops") && (suffix != ".osp")) {
+      std::cerr << "error, infile must be called xxx.ops or xxx.osp\n";
+      exit(1);
+    }
+    std::string base = in.substr(0, in.size() - 4);
+    out = base + DENSE_META_SUFFIX + suffix;
```

Comment:
This suffix seems to be only used when using the converter. I think a converted index should have the exact same data as the same index built freshly in the files named the same (unless both versions have a version field and we can check that in the code).

Reply:
As I read from your Email you would  prefer to always write the dense meta data to an extra file. I will change this.

Reply:
(together with my next PR for prefix  compression I am preparing a index.configuration file (json) where we can also save information about the index in a more explicit way.

Reply:
@joka921 it doesn't really matter if it's one file. It's just that an index upgrade tool like the new MetaDataConvertor should output the same format as would be used when building from scratch with that version.